### PR TITLE
[ENG-6038] Implement FigshareStorageAddonImp

### PR DESCRIPTION
**Pull request by opaduchak**
Implemented FigshareAddonImp

### CODE REVIEW

File: addon_imps/storage/figshare.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        assert project_id.startswith("projects/")
+
+        async with self.network.GET(
+            f"account/{project_id}",
+        ) as response:
+            return Project.from_json(await response.json_content())
+
+
+###
+# module-local helpers
```
Comment (sh-andriy) [inline addon_imps/storage/figshare.py:154]: I wonder if this comment necessary, noticed that you have the same ones later on?

File: addon_service/common/known_imps.py [hunk 2]
```diff
@@ -73,6 +77,8 @@ class AddonImpNumbers(enum.Enum):
     BOX_DOT_COM = 1001
     ZOTERO_ORG = 1002
     S3 = 1003
+    GOOGLE_DRIVE = 1004
```
Comment (sh-andriy) [inline addon_service/common/known_imps.py:80]: I know we have this number for **Mendeley**, I think you can just change this to `1005`, or/and Figshare to `1007`
`MENDELEY = 1004`

File: addon_service/common/network.py [hunk 3]
```diff
@@ -92,6 +92,8 @@ async def _try_send(self, request: HttpRequestInfo):
             request.http_method,
             _url,
             headers=await _private.get_headers(),
+            params=request.query,
+            json=request.json,
             # TODO: content
```
Comment (sh-andriy) [inline addon_service/common/network.py:97]: Are these **TODO's** you currently have doable or if not do we need to include all of them if we won't be doing them?

File: addon_toolkit/interfaces/citation.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ class CitationConfig:
     external_account_id: str | None = None
 
 
-class ItemType(Enum):
+class ItemType(StrEnum):
```
Comment (sh-andriy) [inline addon_toolkit/interfaces/citation.py:34]: I'm curious will this change affect the `Zotero` and `Mendeley API` implementations? If not leave it as you did

**Review by sh-andriy — CHANGES_REQUESTED** [review]
Overall looks pretty good, just need to clarify some points that I commented

File: addon_toolkit/interfaces/citation.py (hunk 4 above)
Comment (opaduchak) [reply addon_toolkit/interfaces/citation.py:34]: This has been done to fix filters invokation from frontend

File: addon_service/common/network.py (hunk 3 above)
Comment (opaduchak) [reply addon_service/common/network.py:97]: These TODO's had been here before me


I'd leave them as there might be need to include urlencoded data or something else.

Comment (sh-andriy) [reply addon_service/common/network.py:97]: Oh okay, if they had been here before than leave it as it is

File: addon_service/common/known_imps.py (hunk 2 above)
Comment (opaduchak) [reply addon_service/common/known_imps.py:80]: This is Google drive related

File: addon_imps/storage/figshare.py (hunk 1 above)
Comment (opaduchak) [reply addon_imps/storage/figshare.py:154]: The same comment was in different file

**Review by sh-andriy — APPROVED** [review]
LGTM

File: addon_toolkit/imp.py [hunk 5]
```diff
@@ -87,7 +87,9 @@ async def invoke_operation(
         if not inspect.iscoroutinefunction(_operation_method):
             _operation_method = sync_to_async(_operation_method)
         _result = await _operation_method(**_kwargs)
-        assert isinstance(_result, operation.result_dataclass)
+        assert isinstance(
+            _result, operation.result_dataclass
+        ), f"expected {operation.result_dataclass.__name__} type to bre returned from method {_operation_method.__name__}, got {_result.__class__.__name__}"
```
Comment (brianjgeiger) [inline addon_toolkit/imp.py:92]: This was mentioned in the Google Drive review, but 
```suggestion
        ), f"expected {operation.result_dataclass.__name__} type to be returned from method {_operation_method.__name__}, got {_result.__class__.__name__}"
```

**Review by brianjgeiger — APPROVED** [review]
Seems good to me

**Pull request merged**